### ISSUE TITLE
RR type encoding length

### ISSUE THREAD

**Issue by pusateri** [mafintosh/dns-packet#33]
Currently, the RR type encoding length is set to the length of the RR type data plus 2. RR type A has length 6, RR Type AAAA has length 18, etc. I think this includes the RDATA LEN short but it seems like it shouldn't.

**Comment by pusateri** [comment]
If you agree this should be changed, I can submit a patch that changes all of the RR types to return the RDLEN and write the RDLEN in question.encode() and answer.encode() instead of in each RR type encoder.

**Comment by silverwind** [comment]
Yes, this seems a bit odd. Do we actually need these `encodingLength` properties? Why not just `buf.length`?

**Comment by pusateri** [comment]
encode() uses them to calculate the buffer length to allocate:
[](https://github.com/mafintosh/dns-packet/blob/eddb7ab7e40002db72b8beef9731477ed8200e9c/index.js#L908)
Which means it runs the same code twice for name and string encoding to calculate the length the first time and to copy into the buffer the second time. This seems like a lot of work to pre-calculate the length of the Buffer. Passing in a buffer to encode() gets around this issue and then none of the encodingLength code is used.

**Comment by pusateri** [comment]
But this is secondary to the problem that the encoders for each RR type include the extra RDLEN bytes.
